ross/rotor_assembly.py

```diff
@@ -1350,6 +1350,215 @@ def _lti(self, speed, frequency=None):
+        sB = (final_speed - initial_speed) / (
+            np.exp(-lambdat * final_time) - np.exp(-lambdat * initial_time)
+        )
+
+        # This code below here is used for acceleration and torque application to the rotor. As of
+        # september/2020 it is unused, but might be implemented in future releases. These would be
+        # run-up and run-down operations and variations of operating conditions.
+        #
+        # sAT = (
+        #     TorqueI * np.exp(-lambdat * final_time) - TorqueF * np.exp(-lambdat * initial_time)
+        # ) / (np.exp(-lambdat * final_time) - np.exp(-lambdat * initial_time))
+        # sBT = (TorqueF - TorqueI) / (
+        #     np.exp(-lambdat * final_time) - np.exp(-lambdat * initial_time)
+        # )
+        #
+        # SpeedV = sA + sB * np.exp(-lambdat * t)
+        # TorqueV = sAT + sBT * np.exp(-lambdat * t)
+        # AccelV = -lambdat * sB * np.exp(-lambdat * t)
+
+        # Get default matrices
+        K = self.K(speed)
+        C = self.C(speed)
+        G = self.G()
+        M = self.M()
+
+        if self.number_dof == 4:
+            Kst = np.zeros((self.ndof, self.ndof))
+        elif self.number_dof == 6:
+            Kst = self.Kst()
+
+        angular_position = (
+            sA * t - (sB / lambdat) * np.exp(-lambdat * t) + (sB / lambdat)
+        )
+
+        self.Omega = sA + sB * np.exp(-lambdat * t)
+        self.AccelV = -lambdat * sB * np.exp(-lambdat * t)
+
+        lambd, ModMat = la.eigh(K, M, type=1, turbo=False)
+
+        wn = np.sqrt(lambd)
+        n_harmonics = 5
+
+        up_freq = np.where(wn >= (n_harmonics * speed))[0].tolist()
+        max_freq = up_freq[0]
+
+        if max_freq % 2:
+            forced_modes = int(max_freq - 1)
+        else:
+            forced_modes = int(max_freq)
+
+        if isinstance(number_modes, str):
+            if number_modes in ["force"]:
+                number_modes = forced_modes
+            else:
+                raise ValueError(
+                    f"The string passed in the number of modes of does not match : 'force'. \nThe value was: {number_modes}"
+                )
+        else:
+            if number_modes > self.ndof:
+                number_modes = self.ndof
+
+            if number_modes < forced_modes:
+                warnings.warn(
+                    f"The number of modes of {number_modes} is lower than the forced one. This could lead to wrong results."
+                )
+
+        self.size = 2 * number_modes
+
+        y0 = np.zeros(self.size)
+
+        ModMat = ModMat[:, :number_modes]
+        ModMat = ModMat
+
+        # Modal transformations
+        self.Mmodal = ((ModMat.T).dot(M)).dot(ModMat)
+        self.Cmodal = ((ModMat.T).dot(C)).dot(ModMat)
+        self.Gmodal = ((ModMat.T).dot(G)).dot(ModMat)
+        self.Kmodal = ((ModMat.T).dot(K)).dot(ModMat)
+        self.Kstmodal = ((ModMat.T).dot(Kst)).dot(ModMat)
+
+        self.Fmodal = (ModMat.T).dot(F.T)
```

Comment:
I think it is easier to ready the code if we use the mat_mul operator (@) for matrix multiplication.

Reply:
Ok